Marketplace/controllers/api/apiProductController.js

```diff
@@ -1,11 +1,77 @@
-const db = require("../../database/models");
-const entidad = db.Producto;
+const productoRepository = require("../../repositories/productoRepository");
+const categoriaRepository = require("../../repositories/categoriaRepository");
 
 module.exports = {
-	listado: (req, res) => {
-
+	listado: async (req, res) => {
+		let listado = null;
+		listado = await productoRepository.ObtenerTodas();
+		// *** PRODUCTOS ***
+		let productos = [];
+		listado.map(n => {
+			let aux=[]
+			n.imagenes.map(m => aux.push(m.ruta));
+			productos.push({
+				id: n.id,
+				nombre: n.nombre,
+				descripcion: n.descripcion,
+				categoria: n.categoria.nombre,
+				imagenes: aux,
+				url: "/api/productos/" + n.id,
+			});
+		});
+		let resumenProductos = {
+			cantidad: listado.length,
+			detalle: productos,
+		};
+		// *** CATEGORÍAS ***
+		listado = await categoriaRepository.ObtenerTodas();
+		let categorias = [];
+		listado.map(n => {
+			let aux = [];
+			n.productos.map(m => {
+				aux.push({
+					id: m.id,
+					nombre: m.nombre,
+					url: "/api/productos/" + m.id,
+				});
+			});
+			categorias.push({
+				id: n.id,
+				nombre: n.nombre,
+				cantidad_de_productos: n.productos.length,
+				productos: aux,
+			});
+		});
+		let resumenCategorias = {
+			cantidad: listado.length,
+			detalle: categorias,
+		};
+		// *** FINAL ***
+		let respuesta = {
+			productos: resumenProductos,
+			categorias: resumenCategorias,
+		};
+		res.json(respuesta);
 	},
-	detalle: (req, res) => {
 
+	detalle: async (req, res) => {
+		let listado = null;
+		listado = await productoRepository.ObtenerPorId(req.params.id);
```

Comment:
Estas dos lineas podrian ser una sola. Ademas en nombre de la variable no es correcta. La funcion del repositorio devuelve un producto, no un listado. ASi seria mas correcto:
```suggestion
		let producto = await productoRepository.ObtenerPorId(req.params.id);
```

Reply:
Christian, 
- A partir de tu comentario, recién le cambié el nombre de _listado_ a _data_.
- Mi intención es estandarizar. Por eso le puse el mismo nombre (data) las 3 veces. Para que quedara en claro dónde se toma la información, más  allá de que todos sabemos. Me parece que así, ayuda a la lectura del código. A mí, por lo menos me ayuda. Así como vos usaste entidad en todos los repositorios, creo que es una manera simple de estandarizar luego el código entre las diferentes tablas.
- Por esa razón también usé el _let_. Si bien como vos decís podría ser una sola línea, y yo soy "pro" eso, lo dejé así para que quedara estandarizada la línea del await. Si se leyera una sola vez en cada método del controller, entonces sí lo haría en una sola línea.